Index: venv/lib/python3.14/site-packages/sqlalchemy/dialects/mysql/base.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/venv/lib/python3.14/site-packages/sqlalchemy/dialects/mysql/base.py b/venv/lib/python3.14/site-packages/sqlalchemy/dialects/mysql/base.py
--- a/venv/lib/python3.14/site-packages/sqlalchemy/dialects/mysql/base.py
+++ b/venv/lib/python3.14/site-packages/sqlalchemy/dialects/mysql/base.py	(date 1770728439170)
@@ -2558,7 +2558,10 @@
 
     def visit_VARCHAR(self, type_: VARCHAR, **kw: Any) -> str:  # type: ignore[override]  # NOQA: E501
         if type_.length is not None:
-            return self._extend_string(type_, {}, "VARCHAR(%d)" % type_.length)
+            if isinstance(type_.length, str) or isinstance(type_.length, sqltypes.String):
+                return self._extend_string(type_, {}, str(type_.length))
+            else:
+                return self._extend_string(type_, {}, "VARCHAR(%d)" % int(type_.length))
         else:
             raise exc.CompileError(
                 "VARCHAR requires a length on dialect %s" % self.dialect.name
